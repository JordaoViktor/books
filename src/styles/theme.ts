export default {
  colors: {
    primary: "#FFFF",
    secondary: "#333333",
    tertiary: "#AB2680",
    background: "#E5E5E5",
    dark: "#000",

    lightenText: "#FFFF",
    darkText: "#333333",

    darkOpacity200: "rgba(0,0,0,0.20)",
    darkOpacity300: "rgba(0,0,0,0.32)",
    darkOpacity400: "rgba(0,0,0,0.4)",

    transparent: "transparent"
  },
  fonts: {
    light: "Heebo_300Light",
    normal: "Heebo_400Regular",
    bold: "Heebo_500Medium",
  },
}