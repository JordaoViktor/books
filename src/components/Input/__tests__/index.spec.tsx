import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import theme from '@styles/theme'
import { Input } from '../index';

interface IProviderProps {
  children?: React.ReactNode,
}

const Provider = ({ children }: IProviderProps) => (
  <ThemeProvider theme={theme} >
    {children}
  </ThemeProvider>
);

describe('Input Component', () => {
  test('matches input snapshot', () => {

    const inputRender = renderer
      .create(
        <ThemeProvider theme={theme} >
          <Input />
        </ThemeProvider>)
      .toJSON();

    expect(inputRender).toMatchSnapshot();

  });

  it('ensure button receives some color', () => {
    const { getByTestId } = render(<Input color="#FAFA" />, { wrapper: Provider })

    const inputComponent = getByTestId('input')
    expect(inputComponent.props.style[0].color).toEqual("#FAFA")
  })

  it('ensure button receives do not some color', () => {
    const { getByTestId } = render(<Input />, { wrapper: Provider })

    const inputComponent = getByTestId('input')
    expect(inputComponent.props.style[0].color).toEqual("#FFFF")
  })
})