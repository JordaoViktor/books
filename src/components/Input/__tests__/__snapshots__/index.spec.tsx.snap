// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component matches input snapshot 1`] = `
<TextInput
  allowFontScaling={true}
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "color": "#FFFF",
        "fontFamily": "Heebo_400Regular",
        "fontSize": 31,
        "height": "100%",
        "width": "100%",
        "zIndex": 3,
      },
    ]
  }
  testID="input"
  underlineColorAndroid="transparent"
/>
`;
